### DISCUSSION TITLE
Potential logic issue in setup_hook method

### DISCUSSION THREAD

**Discussion by 99roblems** [nickpourazima/rotary-phone-audio-guestbook#67] [Ideas]
Hi there,

I've been working with this project and noticed a potential issue in the setup_hook method. The current implementation seems to have reversed logic for NC and NO switches - unless I have my own logic backwards. I've analyzed the code and documentation, and I believe a small change could resolve this issue.

I have an NO hook type phone, and it only seemed to work when I have the config.yml set to NC. It's easy enough to just flip my config, but I think it might be worth testing a minor modification to setup_hook?

I'd be happy to create a pull request for this. I'm also interested in contributing further to this project as time permits, if you're open to new contributors.

Thank you for your time and for maintaining this project!

Best,

Rob

**Comment by nickpourazima** [comment]
Hey @99roblems,

Absolutely, that would be awesome! Thanks for letting me know about the potential bug, I haven't had a chance to test out the latest too extensively. But yes I would gladly welcome any help! Your adjustment might even address #66

**Reply by davidkindred** [reply]
@99roblems Rob, I encountered the same problem and had to reverse the hook type to get it to work. --David
